### ISSUE TITLE
Incorrect number of arguments reported when function is imported over

### ISSUE THREAD

**Issue by MarkCBell** [PyCQA/pylint#3675]
### Steps to reproduce
1. Create `other_file.py` containing:
```
def f(x, y):
    return x + y
```
2. Run pylint on:
```
"""pylint test module"""
    
def f(x):
    return x + 1

def test_one(param):
    """function defined above is matched first"""
    tmp = f
    if param == 0:
        from other_file import f as tmp  # pylint: disable=import-outside-toplevel
        return tmp(1, 4)  # <-- E1121: Too many positional arguments for function call (too-many-function-args)
    return tmp(1)

print(test_one(0))  # Prints "5"
print(test_one(1))  # Prints "2"
```

### Current behavior
Pylint reports that line 12 was passed "too many positional arguments for function call (too-many-function-args)"

### Expected behavior

Pylint should not report an issue since within this if statement tmp refers to the imported function which can be called with two arguments.

### pylint --version output

pylint 2.5.2
astroid 2.4.1
Python 3.8.2 (default, Apr 27 2020, 15:53:34) 
[GCC 9.3.0]

**Comment by MarkCBell** [comment]
The issues appears to be caused by changes introduced in commit https://github.com/PyCQA/pylint/commit/01dfa522195d79217c43065d3d013e2ee31d47b7 to safe_infer in pylint/checkers/utils.py to address https://github.com/PyCQA/pylint/issues/2503. This appears to be because the safe_infer only checks the types of the nodes are the same. In this case they are both functions which is good enough to pass this test but they are functions which take different numbers of arguments and so there is ambiguity about which one to use.

**Comment by Pierre-Sassoulas** [comment]
Thank you for opening the issue and for the analysis.

**Comment by MarkCBell** [comment]
Thank you. As the original title might suggest, I encountered this by running pylint on the following code:
```
import networkx as nx

G = nx.Graph([(1, 2), (2, 3), (2, 4)])
path = nx.algorithms.shortest_path(G, 1, 4)  # <-- Triggers E1121: Too many positional arguments for function call
print(path)
```
This is because networkx uses this trick to import `shortest_path` over an existing function.
